pkg/object/httpserver/mux.go

```diff
@@ -454,7 +456,9 @@ func (m *mux) handleRequestWithCache(rules *muxRules, ctx context.HTTPContext, c
 			path = ci.path.pathRE.ReplaceAllString(path, ci.path.rewriteTarget)
 			ctx.Request().SetPath(path)
 		}
+		m.beforeGlobalHTTPFilterHandle(ctx, rules)
```

Comment:
It seems `beforeGlobalHTTPFilterHandle` `jumpIf: END` will only jump to the end of `beforePipeline`. Which means if `beforePipeline` is used to check authentication, then requests without authentication can still go to following pipelines. 

One way to solve it is to change `HTTPPipeline.Handle` to return if it `jumpif` to end. Another way may be check if context if canceled. I will check the related codes. 
More advices need here.

Reply:
This is a problem. I think we should discuss how to solve it. 
First of all, I also think that if `jumpIf: END` is triggered in `beforePipeline`, `pipeline` and `afterPipeline` should not be executed.
According to the current logic, it is a better choice to check whether the context is cancelled.
More advices need here.

Reply:
I wonder if canceling context is a good idea if the context is also used to send the response after failed authentication? Or could there be a flag `beforePipelineCancelled`?

Here's a bit detailed proposal how to modify `HTTPPipeline.Handle` to pass the `jumpIf: END` information. Currently `HTTPPipeline.Handle` does not make difference between arriving to the end of pipeline from `jumpIf: END` [1](https://github.com/megaease/easegress/blob/main/pkg/object/httppipeline/httppipeline.go#L422-L428) or by just finishing the last filter [2](https://github.com/megaease/easegress/blob/main/pkg/object/httppipeline/httppipeline.go#L407-L409). Maybe the `HTTPPipeline.getNextFilterIndex` could return tuple (int, bool) where the boolean indicates if it jumped to the end. Then this info could be passed all way to the `pkg/object/httpserver/mux.go` when calling `m.beforeGlobalHTTPFilterHandle(ctx, rules)` and the following pipelines could be skipped, if jumped to the END in beforePipeline.

Reply:
I prefer `HTTPPipeline.Handle` to return `LabelEND`, `LabelEND` only used in `HTTPPipeline` to control filter execution flow. And in `HTTPPipeline` it don't check if context is canceled or call cancel function to cancel it. Depend on the filter implementation, in some case it may not call cancel function. So, if we use `HTTPPipeline.Handle`, it can 100% reach end.

Reply:
I agree with you. 
I think `HTTPPipeline` should return a string `lastResult` or `labelEND`. 
And, modify the return value of `HTTPPipeline.getNextFilterIndex`.